### PR TITLE
add support to upscale population to fill in the required fraction

### PR DESCRIPTION
**Pull request by bhavyalatha26**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2419)

### CODE REVIEW

File: src/main/scala/beam/sim/BeamHelper.scala [hunk 1]
```diff
@@ -780,6 +783,118 @@ trait BeamHelper extends LazyLogging {
...
+              val attrValue = scenario.getHouseholds.getHouseholdAttributes.getAttribute(oldhh.getId.toString, attr)
+              scenario.getHouseholds.getHouseholdAttributes.putAttribute(newhh.getId.toString, attr, attrValue)
+            }
+        }
+      }
+      // reset the counter
+      populationCounter = 0
+      logger.info(s"""After sampling:
+                     |Number of households: ${scenario.getHouseholds.getHouseholds.keySet.size}
+                     |Number of vehicles: ${getVehicleGroupingStringUsing(
```
Comment (wrashid) [inline src/main/scala/beam/sim/BeamHelper.scala]: let extract sampling into separate class to save 200+ lines here

Comment (bhavyalatha26) [reply src/main/scala/beam/sim/BeamHelper.scala]: ok sure

Comment (bhavyalatha26) [reply src/main/scala/beam/sim/BeamHelper.scala]: Made the changes @wrashid

**Comment by bhavyalatha26** [comment]
test!

**Comment by bhavyalatha26** [comment]
@wrashid the test are passing now , we can merge the PR

**Pull request merged**